vaultwarden: 1.34.3 -> 1.35.0; vaultwarden.webvault: 2025.7.0.0 -> 2025.12.0

eyJhb opened the pull request:
## Things done
- Updated vaultwarden
- Removed potentially unneeded code from `update.nix`
- Changed webvault to use rev instead of tag, as that's what we pull from the docker settings file
- Fixed build on newest webvault version.

---

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

alvr commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 474836`
Commit: `e6f53f76b15abcdbae2481a0f94de69bce1f716d`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>vaultwarden (vaultwarden-sqlite)</li>
    <li>vaultwarden-mysql</li>
    <li>vaultwarden-postgresql</li>
  </ul>
</details>

dotlambda commented on pkgs/by-name/va/vaultwarden/webvault.nix:
```diff
@@ -9,16 +9,16 @@
     repo = "vw_web_builds";
-    tag = "v${version}";
-    hash = "sha256-CnVOi4xE0+VFTm0yI4++MBv8w0vgtsRE5E2RwsLojMI=";
+    rev = "v${version}";
```
```suggestion
    tag = "v${version}";
```

dotlambda replied:
Thanks!

eyJhb commented on pkgs/by-name/va/vaultwarden/webvault.nix:
```diff
@@ -31,6 +31,11 @@ buildNpmPackage rec {
   };
 
+  preBuild = ''
+    # using sass-embedded fails at executing node_modules/sass-embedded-linux-x64/dart-sass/src/dart
```
It works and compiles just fine, by doing this. This is generally what is done in other packages that has this issue :)

eyJhb committed:
vaultwarden: updated update.nix to correctly find webvault tag

eyJhb committed:
vaultwarden: 1.34.3 -> 1.35.0

eyJhb committed:
vaultwarden.webvault: 2025.7.0.0 -> 2025.12.0

Added a patch to use our own dart-sass, instad of the one provided by sass-embedded

eyJhb commented:
> > I've tested this and am currently running it on my own server without any noticeable issues
> 
> Have you tested the new OIDC functionality?

Indeed. That's why this adventure happened. It works just fine, and I have it configured here https://git.fricloud.dk/fricloud/server-configs/src/branch/main/machines/gerd/services/vaultwarden.nix#L36-L44 :)

alvr commented:
Vaultwarden v1.35.1 has been released. Maybe is possible to upgrade to this version instead:

https://github.com/dani-garcia/vaultwarden/releases/tag/1.35.1

eyJhb commented:
I'll try to do that later, but if it can me merged before that, I think we should just do so, and make a new pr. Whatever comes first basically :)

On 30 December 2025 18.49.14 CET, "Álvaro Salcedo García" ***@***.***> wrote:
>alvr left a comment (NixOS/nixpkgs#474836)
>
>Vaultwarden v1.35.1 has been released. Maybe is possible to upgrade to this version instead:
>
>https://github.com/dani-garcia/vaultwarden/releases/tag/1.35.1
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/NixOS/nixpkgs/pull/474836#issuecomment-3700064738
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>

dotlambda reviewed (approved):
The NixOS test also fails on master.
